### PR TITLE
[EventHubs] update vendor storage blob code

### PR DESCRIPTION
**Pull request by yunhaoling**
trying to resolve issues: https://github.com/Azure/azure-sdk-for-python/issues/16192
use the storage-blob version v12.7.1: https://github.com/Azure/azure-sdk-for-python/tree/azure-storage-blob_12.7.1, commit id: e8edeb85bcb9f92cb3c6299b893eb99be55bdd51

### CODE REVIEW

**Review by swathipil — APPROVED** [review]
lgtm

File: sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/azure/eventhub/extensions/checkpointstoreblobaio/_vendor/storage/blob/_shared/avro/__init__.py [hunk 1]
```diff
@@ -1,13 +1,5 @@
-# coding=utf-8
-# --------------------------------------------------------------------------
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/azure/eventhub/extensions/checkpointstoreblobaio/_vendor/storage/blob/_shared/avro/__init__.py:2]: This is a bit weird - but I think it's just a Github diff issues...

File: sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/azure/eventhub/extensions/checkpointstoreblobaio/_vendor/storage/blob/_upload_helpers.py [hunk 2]
```diff
@@ -147,8 +150,9 @@ def upload_block_blob(  # pylint: disable=too-many-locals
             validate_content=validate_content,
             headers=headers,
             tier=tier.value if tier else None,
+            blob_tags_string=blob_tags_string,
             **kwargs)
-    except StorageErrorException as error:
+    except HttpResponseError as error:
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/azure/eventhub/extensions/checkpointstoreblobaio/_vendor/storage/blob/_upload_helpers.py:155]: Given that there's been some updates to error handling - do any of out live tests deal with errors raised from Storage? It shouldn't be an issue, but would be nice to know we have coverage :)

**Review by annatisch — APPROVED** [review]
A couple of questions but generally looks good to me.

File: sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/azure/eventhub/extensions/checkpointstoreblobaio/_vendor/storage/blob/_shared/avro/__init__.py (hunk 1 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/azure/eventhub/extensions/checkpointstoreblobaio/_vendor/storage/blob/_shared/avro/__init__.py:2]: the latest storage blob doesn't have this file any more, let's pretend it doesn't exist 😄

File: sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/azure/eventhub/extensions/checkpointstoreblobaio/_vendor/storage/blob/_upload_helpers.py (hunk 2 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/azure/eventhub/extensions/checkpointstoreblobaio/_vendor/storage/blob/_upload_helpers.py:155]: nope, we don't have tests covering storage errors as for most blob storage errors, our sdk simply bubble up or ignore them.

The only errors we care about are `ResourceModifiedError, ResourceExistsError` which are treated as `OwnershipLostError`:
https://github.com/Azure/azure-sdk-for-python/blob/f0697ff8b1bcafbd1a82e7961098f14229295e2b/sdk/eventhub/azure-eventhub-checkpointstoreblob/azure/eventhub/extensions/checkpointstoreblob/_blobstoragecs.py#L163-L173


I don't think we need to cover all of the blob storage errors now based on our current "bubble up or ignore" approach. but I'm always open to adding new tests if needed :P

**Pull request merged**